src/tool.cpp

```diff
@@ -110,40 +107,43 @@ DigParams getDigParams(const ItemGroupList &groups,
 	std::string result_main_group = "";
 
 	int level = itemgroup_get(groups, "level");
-	//infostream<<"level="<<level<<std::endl;
 	for (ToolGCMap::const_iterator i = tp->groupcaps.begin();
 		 	i != tp->groupcaps.end(); ++i) {
-		const std::string &name = i->first;
-		//infostream<<"group="<<name<<std::endl;
 		const ToolGroupCap &cap = i->second;
-		int rating = itemgroup_get(groups, name);
+
+		int leveldiff = cap.maxlevel - level;
+		if (leveldiff < 0)
+			continue;
+
+		const std::string &groupname = i->first;
 		float time = 0;
+		int rating = itemgroup_get(groups, groupname);
 		bool time_exists = cap.getTime(rating, &time);
-		if(!result_diggable || time < result_time){
-			if(cap.maxlevel >= level && time_exists){
-				result_diggable = true;
-				int leveldiff = cap.maxlevel - level;
-				result_time = time / MYMAX(1, leveldiff);
-				if(cap.uses != 0)
-					result_wear = 1.0 / cap.uses / pow(3.0, (double)leveldiff);
-				else
-					result_wear = 0;
-				result_main_group = name;
-			}
+		if (!time_exists)
+			continue;
+
+		if (leveldiff > 1)
+			time /= leveldiff;
+		if (!result_diggable || time < result_time) {
+			result_time = time;
+			result_diggable = true;
+			if (cap.uses != 0)
+				result_wear = 1.0 / cap.uses / pow(3.0, leveldiff);
+			else
+				result_wear = 0;
+			result_main_group = groupname;
 		}
 	}
-	//infostream<<"result_diggable="<<result_diggable<<std::endl;
-	//infostream<<"result_time="<<result_time<<std::endl;
-	//infostream<<"result_wear="<<result_wear<<std::endl;
 
-	if(time_from_last_punch < tp->full_punch_interval){
+	if (time_from_last_punch < tp->full_punch_interval) {
 		float f = time_from_last_punch / tp->full_punch_interval;
-		//infostream<<"f="<<f<<std::endl;
-		result_time /= f;
-		result_wear /= f;
+		if (f > 1.0) {
```

Comment:
You're right. Anyway, l just noticed (```$ ag -r -A2 getDigParams```) that it's unused code.
l'll remove it in a separate commit.